pkg/k8s/statefulsets.go

```diff
@@ -42,11 +45,17 @@ func WatchStatefulSets() error {
 
 	// Watch for events on the watcher channel
 	for event := range watcher.ResultChan() {
+		// Check if the event object is of type *v1.StatefulSet
 		if statefulSet, ok := event.Object.(*v1.StatefulSet); ok {
-			//log.Info("StatefulSet containers: ", statefulSet.Spec.Template.Spec.Containers)
+			if !ok {
+				// If it's not a StatefulSet, log a warning and continue with the next event.
+				log.Warn("Received an event that is not a StatefulSet. Skipping this resource...")
+				continue
+			}
 
-			// check if the node is DA, if so, send it to the queue to generate the multi address
-			if strings.HasPrefix(statefulSet.Name, "da") {
+			// Check if the StatefulSet is valid based on the conditions
+			if isStatefulSetValid(statefulSet) {
+				// Perform necessary actions, such as adding the node to the Redis queue
```

Comment:
I do not have enough context to review the logic. Just one comment: we can remove a lot of those comments as the code is descriptive enough.